src/scripts/main.js

```diff
@@ -1,3 +1,259 @@
+
+// #region creating a form (constructor)
+function newForm(atributes, cities) {
+  const form = document.createElement('form');
+
+  for (const [key, value] of Object.entries(atributes)) {
+    const label = document.createElement('label');
+
+    if (value === 'Office') {
+      const select = document.createElement('select');
+
+      label.textContent = `${value}:`;
+      select.setAttribute('data-qa', `${key}`);
+      select.name = key;
+
+      cities.forEach((city) => {
+        const option = document.createElement('option');
+
+        option.value = city;
+        option.textContent = city;
+
+        select.appendChild(option);
+      });
+
+      label.appendChild(select);
+      form.appendChild(label);
+    } else {
+      const input = document.createElement('input');
+
+      label.textContent = `${value}:`;
+      input.setAttribute('data-qa', `${key}`);
+      input.name = `${key}`;
+      input.type = 'text';
+      input.setAttribute('required', '');
+
+      label.appendChild(input);
+      form.appendChild(label);
+    }
+  }
+
+  const button = document.createElement('button');
+
+  button.textContent = 'Save to table';
+  button.type = 'submit';
+
+  form.appendChild(button);
+
+  return form;
+}
+
+// #endregion
+
+// #region form submit event
+employee.querySelector('button').addEventListener('click', (e) => {
+  const inputName = employee.querySelector('[data-qa="name"]');
+  const inputPosition = employee.querySelector('[data-qa="position"]');
+  const inputOffice = employee.querySelector('[data-qa="office"]');
+  const inputAge = employee.querySelector('[data-qa="age"]');
+  const inputSalary = employee.querySelector('[data-qa="salary"]');
+
+  const valuesOfInputs = [
+    inputName,
+    inputPosition,
+    inputOffice,
+    inputAge,
+    inputSalary,
+  ];
+
+  if (
+    inputName.value.trim() !== '' &&
+    inputName.value.length >= 4 &&
+    inputAge.value.trim() !== '' &&
+    parseInt(inputAge.value) >= 18 &&
+    parseInt(inputAge.value) <= 90 &&
+    inputPosition.value !== '' &&
+    inputSalary.value !== ''
+  ) {
+    const row = document.createElement('tr');
+
+    valuesOfInputs.forEach((input) => {
+      const cell = document.createElement('td');
+
+      if (input.getAttribute('data-qa') === 'salary') {
+        cell.textContent = formatCurrency(input.value);
+      } else {
+        cell.textContent = input.value;
+      }
+
+      row.appendChild(cell);
+    });
+
+    tbody.appendChild(row);
+
+    notificationMessage('Adding successfully', 'success');
+
+    employee.reset();
+  } else {
+    if (inputName.value.length < 4) {
+      notificationMessage('Name length is less than 4 characters', 'error');
+    }
+
+    if (parseInt(inputAge.value) < 18) {
+      notificationMessage('Your age is too young :(', 'error');
+    }
+
+    if (parseInt(inputAge.value) > 90) {
+      notificationMessage('Your age is too old :(', 'error');
+    }
+
+    if (inputPosition.value.trim() === '') {
```

Comment:
```suggestion
    if (!inputPosition.value.trim()) {
```
shorter version

But also, you should show the error if age and salary inputs are empty. Now when I click Save button without these inputs nothing happens:

![image](https://github.com/user-attachments/assets/be05eae5-d032-41c5-b30a-5c9bf2818b03)